src/RCall.jl

```diff
@@ -29,13 +30,24 @@ export RObject,
    robject, rcopy, rparse, rprint, reval, rcall, rlang,
    rimport, @rimport, @rlibrary, @rput, @rget, @var_str, @R_str
 
-const depfile = joinpath(dirname(@__FILE__),"..","deps","deps.jl")
-if isfile(depfile)
-    include(depfile)
+@static if @has_preference("Rhome") || @has_preference("libR")
```

Comment:
ah this `@static` block means you need to do another `precompile` if you go from global R to preferences-R. But if the RCall source hasn't changed, then this won't work. Likewise, changing the preferences-R path won't be reflected. It's probably better to handle some of this logic as part of `__init__`: it will increase load-time slightly (but then again, you still have to start R afterwards, so a few milliseconds finding a path dynamically is nothing).

Reply:
I don't think that's needed. Have you tested this? I have a test script at https://gist.github.com/frankier/0ac468305f8b04d98c90551eda0da15d you can try. Just set RCALL_PATH to your checkout of this PR and run in from an empty directory. I see the following output:

```
...
Should be Conda R
/home/frankier/.julia/conda/3/lib/R/lib/libR.so
...
Should be CondaPkg R
/path/to/mytest/testproj/.CondaPkg/env/lib/R/lib/libR.so
...
```

The reason this works is whenever we read from a preference at compile time, it is marked as a compile time preference. I think this means that all bytecodes are then keyed off the preference value. The "global install" is keyed as `Rhome = libR = unset`. Then as soon as we set them we get a new key for our bytecode which triggers precompilation. I may have got the details wrong, but that's what I have surmised from the documentation.

Reply:
:thinking: It still feels weird to have a `@static` call for something that's dynamic. My preference would be drop it here because 
- it feels weird to me
- `@static` at the top-level of a module for something that's not platform dependent, doesn't actually get you that much (because the module definition is only evaluated once unless you do something fancy like explicitly invoking `Base.require` or wonky with Revise.jl)

But that's just my preference.

Reply:
That's a good point! I'm not sure why I set that, except I wanted to make extra-sure things would be at compile time, but as you say there's no need and as a result it's a bit odd. I have added a comment about what is going on here and pulled them out into temporary variables to defeat short-circuiting.